lib/em/protocols/linetext2.rb

```diff
@@ -57,6 +57,7 @@ def receive_data data
         while remaining_data.length > 0
           if @lt2_mode == :lines
             if ix = remaining_data.index( @lt2_delimiter )
+              delimiter_length = remaining_data.slice(@lt2_delimiter).length
```

Comment:
If `remaining_data.index(@l2_delimiter)` above is executing the Regexp, then we probably don't want to execute it a second time to get the length of the match; I'm thinking what an alternate mechanism might look like to execute the Regexp just once per chunk.